src/hugr/rewrite/replace.rs

```diff
@@ -616,7 +616,7 @@ mod test {
         entry: bool,
     ) -> Result<BasicBlockID, BuildError> {
         let op: OpType = op.into();
-        let op_sig = op.signature();
+        let op_sig = op.signature().expect("dataflow op needs signature");
```

Comment:
Hmmmm, we can't do something like `impl Into<DataflowOpTrait>` can we? Then we'd know we have a signature. I guess nothing to worry about in a test but generally it feels like it'd be good to try to know statically when we have an Op(Type) with a signature

Reply:
done, I've used this logic in ~3 places, there are probably more. (Could be a follow up task for builder to require `DataflowOpTrait` in more places rather than `Into<OpType>`?)